app/controllers/admin/listing_shapes_controller.rb

```diff
@@ -78,6 +78,18 @@ def update
 
     if update_result.success
       flash[:notice] = t("admin.listing_shapes.edit.update_success", shape: pick_translation(shape[:name]))
+
+      # Onboarding wizard step recording
+      state_changed = Admin::OnboardingWizard.new(@current_community.id)
+        .update_from_event(:listing_shape_updated, update_result)
```

Comment:
We only fire this event from the success branch so we might as well pass directly `update_result.data` in to event handler so it wouldn't have to deal with the Result-object wrapping.